### PR TITLE
Upgrade Kubernetes lib to 1.25 - 1.X

### PR DESCRIPTION
**Pull request by outscale-mdr**
This PR is to upgrade kubernetes lib to 1.25 which consists in:
- upgrade go version to 1.19: [kubernetes PR](https://github.com/kubernetes/kubernetes/pull/115013)
- upgrade ginkgo to v2: [kubernetes PR](https://github.com/kubernetes/kubernetes/pull/109111)
- change the kuberneets image repository (**not mandatory**): [kubernetes PR](https://github.com/kubernetes/kubernetes/pull/109938)
- set pod enforcement policy in e2e tests: [links](https://github.com/kubernetes/enhancements/tree/master/keps/sig-auth/2579-psp-replacement#ga)
- Change wait method in e2e tests: [Kubernetes links](https://github.com/kubernetes/kubernetes/pull/113135)

### CODE REVIEW

File: .github/workflows/e2e_test.yml [hunk 1]
```diff
@@ -59,7 +60,7 @@ jobs:
         osc_region: ${{ secrets.OSC_REGION }}
         image_id: ${{ secrets.OMI_ID }}
         rke_version: "v1.4.2"
-        kubernetes_version: "v1.23.15-rancher1-1"
+        kubernetes_version: "v1.24.9-rancher1-1"
```
Comment (outscale-vbr) [inline .github/workflows/e2e_test.yml:63]: You are upgrading in 1.25 but rancher is stuck at kubernetes 1.24 (and current release of kubernetes is 1.26).

File: tests/e2e/dynamic_provisioning.go [hunk 2]
```diff
@@ -45,6 +46,7 @@ const OSC_REGION = "OSC_REGION"
 
 var _ = Describe("[bsu-csi-e2e] [single-az] Dynamic Provisioning", func() {
 	f := framework.NewDefaultFramework("bsu")
+	f.NamespacePodSecurityEnforceLevel = admissionapi.LevelPrivileged
```
Comment (outscale-vbr) [inline tests/e2e/dynamic_provisioning.go:49]: Ok nice
https://linuxera.org/working-with-pod-security-standards/

File: tests/e2e/testsuites/testsuites.go [hunk 3]
```diff
@@ -478,7 +478,7 @@ func (t *TestDeployment) DeletePodAndWait() {
 		return
 	}
 	e2elog.Logf("Waiting for pod %q in namespace %q to be fully deleted", t.podName, t.namespace.Name)
-	err = e2epod.WaitForPodNoLongerRunningInNamespace(t.client, t.podName, t.namespace.Name)
+	err = e2epod.WaitForPodNotFoundInNamespace(t.client, t.podName, t.namespace.Name, 3*time.Minute)
```
Comment (outscale-vbr) [inline tests/e2e/testsuites/testsuites.go:481]: How did you choice 3 Minute ?

File: osc-bsu-csi-driver/values.yaml [hunk 4]
```diff
@@ -32,7 +32,7 @@ backoff:
 
 sidecars:
   provisionerImage:
-    repository: k8s.gcr.io/sig-storage/csi-provisioner
+    repository: registry.k8s.io/sig-storage/csi-provisioner
```
Comment (outscale-vbr) [inline osc-bsu-csi-driver/values.yaml:35]: This explain why 
https://kubernetes.io/blog/2022/11/28/registry-k8s-io-faster-cheaper-ga/.

File: .github/workflows/e2e_test.yml (hunk 1 above)
Comment (outscale-mdr) [reply .github/workflows/e2e_test.yml:63]: Yes, between 1.24 and 1.25, the API for the CSI is untouched.

And yes, rancher does not have a 1.25 kubernetes cluster => stay on 1.24

File: tests/e2e/dynamic_provisioning.go (hunk 2 above)
Comment (outscale-mdr) [reply tests/e2e/dynamic_provisioning.go:49]: For the test, I intentionnaly did not look for a hardened security, we are free to do whatever we want

**Review by outscale-vbr — APPROVED** [review]

**Review by jerome-jutteau — APPROVED** [review]

**Pull request merged**